budgetApp1/budgetApp1/BudgetViewController.swift

```diff
@@ -9,32 +9,71 @@ import UIKit
 
 class BudgetViewController: UIViewController {
     
+    // MARK: - UISpecs Dependency
+    private let uiSpecs = UISpecs()
+    
     // MARK: - IBOutlets
     @IBOutlet private weak var budgetNameLabel: UILabel!
     @IBOutlet private weak var endDateLabel: UILabel!
     @IBOutlet private weak var startDateLabel: UILabel!
+    @IBOutlet weak var toDateLabel: UILabel!
+    @IBOutlet weak var recentTransactionsLabel: UILabel!
+    @IBOutlet weak var tableView: UITableView!
     
-    private lazy var viewModel = BudgetViewModel(repository: BudgetRepository(), delegate: self)
+    private lazy var budgetViewModel = BudgetViewModel(repository: BudgetRepository(), delegate: self)
```

Comment:
Okay this is a bit worrying. BudgetViewController should only be talking to budgetViewModel. 
Why is there another viewModel involved for this screen?

Reply:
something it too coupled

Reply:
So this is happening because I have a tableView that returns the most recent transactions on the Budget Overview Screen hence I use view models.

Reply:
please don't mark as resolved until the person replies to the comment.
In future, you should have 1 viewModel per screen.
something like budgetViewModel that handles everything. 
What you have now is tightly coupling 2 viewModels to 1 screen